mlflow/models/evaluation/__init__.py

```diff
@@ -0,0 +1,399 @@
+        """
+        return self._location
+
+
+class EvaluationResult:
+    def __init__(self, metrics, artifacts):
+        self._metrics = metrics
+        self._artifacts = artifacts
+
+    @classmethod
+    def load(cls, path):
+        """Load the evaluation results from the specified local filesystem path"""
+        with open(os.path.join(path, "metrics.json"), "r") as fp:
+            metrics = EvaluationMetrics(json.load(fp))
+
+        with open(os.path.join(path, "artifacts_metadata.json"), "r") as fp:
+            artifacts_metadata = json.load(fp)
+
+        artifacts = {}
+
+        artifacts_dir = os.path.join(path, 'artifacts')
+
+        for artifact_name, meta in artifacts_metadata:
+            location = meta["location"]
+            ArtifactCls = load_class(meta["class_name"])
+            content = ArtifactCls.load_content_from_file(
+                os.path.join(artifacts_dir, artifact_name)
+            )
+            artifacts[artifact_name] = ArtifactCls(location=location, content=content)
+
+        return EvaluationResult(metrics=metrics, artifacts=artifacts)
+
+    def save(self, path):
+        """Write the evaluation results to the specified local filesystem path"""
+        os.makedirs(path, exist_ok=True)
+        with open(os.path.join(path, "metrics.json"), "w") as fp:
+            json.dump(self.metrics, fp)
+
+        artifacts_metadata = {
+            artifact_name: {
+                "location": artifact.location,
+                "class_name": _get_fully_qualified_class_name(artifact),
+            }
+            for artifact_name, artifact in self.artifacts.items()
+        }
+        with open(os.path.join(path, "artifacts_metadata.json"), "w") as fp:
+            json.dump(artifacts_metadata, fp)
+
+        artifacts_dir = os.path.join(path, 'artifacts')
+        os.mkdir(artifacts_dir)
+
+        for artifact_name, artifact in self.artifacts.items():
+            artifact._save_content_to_file(
+                os.path.join(artifacts_dir, artifact_name)
+            )
+
+    @property
+    def metrics(self) -> EvaluationMetrics:
+        """
+        A dictionary mapping scalar metric names to scalar metric values
+        """
+        return self._metrics
+
+    @property
+    def artifacts(self) -> Dict[str, EvaluationArtifact]:
+        """
+        A dictionary mapping standardized artifact names (e.g. "roc_data") to
+        artifact content and location information
+        """
+        return self._artifacts
+
+
+class EvaluationDataset:
+    """
+    Represents an input dataset for model evaluation. This is intended for
+    use with the `mlflow.evaluate()`API.
+    """
+
+    NUM_SAMPLE_ROWS_FOR_HASH = 5
+
+    def __init__(self, data, labels, name=None, path=None):
+        """
+        :param data: One of the following:
+         - A numpy array or list of evaluation features, excluding labels.
+         - A Pandas DataFrame, or the path to a serialized DataFrame,
```

Comment:
Can we add support for Spark DataFrames? The default evaluator should call `limit(10000)` and print a warning indicating that only the first 10000 rows are examined.